grammars/python.cson

```diff
@@ -512,7 +512,7 @@
     'captures':
       '1':
         'name': 'storage.type.function.python'
-    'match': '\\b(def|lambda)\\b'
+    'match': '\\b(def|lambda|async def)\\b'
```

Comment:
The `async` keyword should be given the scope `storage.modifier.python`. The way you've implemented it `async def` is given the scope `storage.type.function.python`. In addition if someone adds two spaces between `async` and `def` as opposed to one (for whatever silly reason), your change will not work.

I suggest taking a look at how e.g. language-javascript implements async and await:
https://github.com/atom/language-javascript/blob/master/grammars/javascript.cson

Do note that the javascript grammar is generally much better written, but also more complex, so the "translation" effort will likely not be trivial, although not too difficult since Python's syntax is much simpler. Let us know if you need any assistance. /cc @atom/feedback